addons/sms_twilio/tools/sms_api.py

```diff
@@ -0,0 +1,90 @@
+import logging
+import requests
+
+from odoo import _
+from odoo.exceptions import UserError
+
+from odoo.addons.sms.tools.sms_api import SmsGenericApi
+from odoo.addons.sms_twilio.tools.sms_tools import get_twilio_from_number, get_twilio_status_callback_url
+
+_logger = logging.getLogger(__name__)
+
+
+class SmsTwilioApi(SmsGenericApi):
+    PROVIDER_TO_SMS_FAILURE_TYPE = SmsGenericApi.PROVIDER_TO_SMS_FAILURE_TYPE | {
+        'twilio_authentication': 'sms_twilio_authentication',
+    }
+
+    def __init__(self, env):
+        super().__init__(env)
+
+    def _sms_twilio_send_request(self, to_number, body, uuid):
+        company_sudo = self.env.company.sudo()
+        company_sudo._assert_twilio_sid()
+        from_number = get_twilio_from_number(company_sudo, to_number)
+        data = {
+            'From': from_number.number,
+            'To': to_number,
+            'Body': body,
+            'StatusCallback': get_twilio_status_callback_url(company_sudo, uuid),
+        }
+        try:
+            return requests.post(
+                f'https://api.twilio.com/2010-04-01/Accounts/{company_sudo.sms_twilio_account_sid}/Messages.json',
+                data=data,
+                auth=(company_sudo.sms_twilio_account_sid, company_sudo.sms_twilio_auth_token),
+                timeout=5,
+            )
+        except requests.exceptions.RequestException as e:
+            _logger.warning('Twilio SMS API error: %s', str(e))
+        return None
+
+    def _send_sms_batch(self, messages, delivery_reports_url=False):
+        """ Send SMS using IAP in batch mode
```

Comment:
```suggestion
        """ Send a batch of SMS using twilio.
```

Also, it may be worth  
* specifying that `delivery_reports_url` isn't used here, or
* switch the parameter to `**kwargs` and take `delivery_reports_url` from there in iap (perhaps better in master)